### PR TITLE
[PRO-897] Implement Tool: InspectPoint

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/PRO-897/implement-tool-inspectpoint

### CODE REVIEW

File: packages/replay-data/src/recording-data/DependencyGraph.ts [hunk 1]
```diff
@@ -82,10 +81,10 @@ export default class DependencyGraph {
 
     const normalizedFrames = frames
       .map<RawRichStackFrame | null>(frame => this.normalizeFrameForRichStack(frame))
-      .filter(Boolean);
+      .filter(v => !!v);
```
Comment (Domiii) [inline packages/replay-data/src/recording-data/DependencyGraph.ts:84]: Why?

File: packages/replay-data/src/recording-data/DependencyGraph.ts [hunk 2]
```diff
@@ -120,10 +119,8 @@ export default class DependencyGraph {
  * Cull before getting extra data.
  */
 function shouldIncludeRawRichStackFrame(frame: RawRichStackFrame) {
-  if (!TargetDGEventCodes.includes(frame.kind as any)) {
-    return false;
-  }
-  return true;
+  const includedEventCodes: readonly RichStackFrameKind[] = TargetDGEventCodes;
```
Comment (Domiii) [inline packages/replay-data/src/recording-data/DependencyGraph.ts:122]: Ah!

File: packages/source-parser/src/tree-sitter-setup.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+const LANGUAGE_EXTENSIONS = {
+  javascript: ["js", "jsx", "mjs", "cjs"],
+  html: ["html", "htm"],
+} as const;
+
+const LanguagesByFileExtension = new Map<string, Language>([
+  ...LANGUAGE_EXTENSIONS.javascript.map(ext => [ext, JavaScript] as const),
+  ...["ts"].map(ext => [ext, TypeScript.typescript] as const),
+  ...["tsx"].map(ext => [ext, TypeScript.tsx] as const),
```
Comment (Andarist) [inline packages/source-parser/src/tree-sitter-setup.ts]:
```suggestion
  ...["tsx", "mts", "cts"].map(ext => [ext, TypeScript.tsx] as const),
```

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "configurations": [
```
Comment (Domiii) [inline .vscode/launch.json:2]: NOTE: This is the config to run any file or test using the VSCode debugger.

File: .vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        "${file}"
+      ],
+      "cwd": "${workspaceFolder}",
+      "console": "integratedTerminal"
+    },
+    {
+      "name": "Run $file",
+      "type": "node",
+      "request": "launch",
+      "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/tsx",
```
Comment (Andarist) [inline .vscode/launch.json:22]: It's a cool shortcut. For this use case I usually just open a [JavaScript Debug Terminal](https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_javascript-debug-terminal) and run a script manually. It magically auto-attaches to the process and even (some?) children processes

**Pull request merged**